types/agent.go

```diff
@@ -76,6 +78,7 @@ type AgentsParams struct {
 	Last   *uint64
 	Before *string
 	Name   *string
+	Tags   *[]string
```

Comment:
Oh, also what does it mean to pass a list of many tags?
Is it an `AND` or an `OR`?

I mean, a user could ask for all agents that have the tags `foo` **and** `bar`.
Or a user could ask for all the agents that either have the tag `foo` **or** `bar`.

Or a user could ask (front-end) for a complex query like `(tag=foo && tag=bar) || tag=baz`. How will we able to provide an API that supports that? 🤔 

Maybe we should allow a single `tag` string to do the filtering.
Then in a second future iteration we could add some kind of simple expresion parsing to this tag string to support that kind of list matching.

Reply:
CC @niedbalski

Reply:
This was intended to be the first implementation for AND listing, I agree with @nicolasparada the type should be a string not and array as it doesn't allows future NOT, negation or any other sub-expressions, that string will be parseable by the API, for now  only AND operator and can be separated either by commas or AND  (I would prefer AND) https://github.com/calyptia/cloud/issues/401#issuecomment-1075018109

Reply:
I don't know if supporting `AND` yet. What about quotes and stuff.
There are edge cases we haven't talked over yet.
I suggest just filtering by one single tag for now.

Reply:
> 

something fancier to trim to [a-z]+ and split by AND?

Reply:
sorry, guys i couldn't visualize the concern. i understood that the structure itself is an array and to send it through the client i created this other two functions associated with the type (on another branch for now):

```go
func (a *AgentsParams) TagsAsString() string {
	return strings.Join(*a.Tags, ",")
}

func (a *AgentsParams) TagFilter(tags ...string) {
	a.Tags = &tags
}
```
thinking to accomplish:
```go
opt.TagFilter("value1", "value2")
```


and using it to set on url params like

```go
if params.Tags != nil {
q.Set("tags", params.TagsAsString())
}
```

i couldnt understand how if its "or", "and" affects on client level, i thought that its a problem on repository level